pkg/k8shandler/services.go

```diff
@@ -16,24 +16,34 @@ import (
 func CreateOrUpdateServices(dpl *v1alpha1.Elasticsearch) error {
 	elasticsearchClusterSvcName := fmt.Sprintf("%s-%s", dpl.Name, "cluster")
 	elasticsearchRestSvcName := dpl.Name
+	metricsSvcName := fmt.Sprintf("%s-%s", dpl.Name, "metrics")
 	owner := asOwner(dpl)
 
 	labelsWithDefault := appendDefaultLabel(dpl.Name, dpl.Labels)
 
-	err := createOrUpdateService(elasticsearchClusterSvcName, dpl.Namespace, dpl.Name, "cluster", 9300, selectorForES("es-node-master", dpl.Name), labelsWithDefault, true, owner)
+	err := createOrUpdateService(elasticsearchClusterSvcName, dpl.Namespace, dpl.Name, "cluster", 9300, selectorForES("es-node-master", dpl.Name), nil, labelsWithDefault, true, owner)
 	if err != nil {
 		return fmt.Errorf("Failure creating service %v", err)
 	}
 
-	err = createOrUpdateService(elasticsearchRestSvcName, dpl.Namespace, dpl.Name, "restapi", 9200, selectorForES("es-node-client", dpl.Name), labelsWithDefault, false, owner)
+	err = createOrUpdateService(elasticsearchRestSvcName, dpl.Namespace, dpl.Name, "restapi", 9200, selectorForES("es-node-client", dpl.Name), nil, labelsWithDefault, false, owner)
 	if err != nil {
 		return fmt.Errorf("Failure creating service %v", err)
 	}
+
+	annotations := map[string]string{
+		"service.alpha.openshift.io/serving-cert-secret-name": metricsSvcName,
+	}
+	err = createOrUpdateService(metricsSvcName, dpl.Namespace, dpl.Name, "metrics", 6000, selectorForES("es-metrics", dpl.Name), annotations, labelsWithDefault, false, owner)
```

Comment:
can we actually reformat this to be multi-line and define the annotation inline?